fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -232,7 +232,14 @@ private PlanNode createQueryPlan(QueryStmt stmt, Analyzer analyzer, long default
             throws UserException {
         long newDefaultOrderByLimit = defaultOrderByLimit;
         if (newDefaultOrderByLimit == -1) {
-            newDefaultOrderByLimit = 65535;
+            if (analyzer.getContext().getSessionVariable().defaultOrderByLimit <= -1) {
+                newDefaultOrderByLimit = Long.MAX_VALUE;
+                LOG.warn("The current default Order By Limit number is the maximum value of Long.");
+            } else {
+                newDefaultOrderByLimit = analyzer.getContext().getSessionVariable().defaultOrderByLimit;
```

Comment:
you can just call `analyzer.getContext().getSessionVariable().defaultOrderByLimit` once

Reply:
has been completed